design/repo_maintenance_job_config.md

```diff
@@ -0,0 +1,216 @@
+This design reuses the data structure introduced by design [node-agent affinity configuration](Implemented/node-agent-affinity.md) to make the repository maintenance job can choose which node running on.
+
+## Goals
+- Unify the repository maintenance Job configuration at one place.
+- Let user can choose repository maintenance Job running on which nodes.
+- Replace the existing `velero server` parameters `--maintenance-job-cpu-request`, `--maintenance-job-mem-request`, `--maintenance-job-cpu-limit` and `--maintenance-job-mem-limit` by the proposal ConfigMap.
+
+## Non Goals
+- There was an [issue](https://github.com/vmware-tanzu/velero/issues/7911) to require the whole Job's PodSpec should be configurable. That's not in the scope of this design.
+- Please notice this new configuration is dedicated for the repository maintenance. Repository itself configuration is not covered.
+
+
+## Compatibility
+v1.14 uses the `velero server` CLI's parameter to pass the repository maintenance job configuration.
+In v1.15, those parameters are removed, including `--maintenance-job-cpu-request`, `--maintenance-job-mem-request`, `--maintenance-job-cpu-limit` and `--maintenance-job-mem-limit`.
+Instead, the parameters are read from the ConfigMap labelled as `velero.io/config-map-type=repo-maintenance-job-config`.
+
+## Design
+This design introduces a new ConfigMap labelled as `velero.io/config-map-type=repo-maintenance-job-config` as the source of the repository maintenance job configuration. 
+If the ConfigMap doesn't exist, the internal default values are used.
+
+**Notice**
+* Velero doesn't own this ConfigMap. If the user wants to customize the repository maintenance job, the user needs to create this ConfigMap.
+* Velero reads this ConfigMap content at starting a new repository maintenance job, so the ConfigMap change will take affect until the next created job.
+
+### Structure
+The data structure for ```repo-maintenance-job-config``` is as below:
+```go
+type MaintenanceConfigMap map[string]Configs
+
+type Configs struct {
+    // LoadAffinity is the config for data path load affinity.
+    LoadAffinity []*LoadAffinity `json:"loadAffinity,omitempty"`    
+
+    // The repository maintenance job CPU request setting
+	CPURequest          string `json:"cpuRequest,omitempty"`
+
+    // The repository maintenance job memory request setting
+	MemRequest          string `json:"memRequest,omitempty"`
+
+    // The repository maintenance job CPU limit setting
+	CPULimit            string `json:"cpuLimit,omitempty"`
+
+    // The repository maintenance job memory limit setting
+	MemLimit            string `json:"memLimit,omitempty"`
+}
+
+type LoadAffinity struct {
+    // NodeSelector specifies the label selector to match nodes
+    NodeSelector metav1.LabelSelector `json:"nodeSelector"`
+}
+```
+
+The ConfigMap content is a map.
+If there is a key value as `global` in the map, the key's value is applied to all BackupRepositories maintenance jobs.
```

Comment:
We should clarify that the global key's values are only applied to jobs which don't have their own specific configuration. I see that this is clarified further down in the detailed design, but it might be worth adding to this sentence to make it clearer here too.